src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+
+    private final Map<FruitTransaction.Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return handlers.get(operation);
```

`getHandler` currently returns `handlers.get(operation)` which can return null when `operation` is null or when there is no handler for that operation. Per the requirements the strategy must provide an applicable handler or fail fast with an informative message. Change this method to validate `operation` (throw `new RuntimeException("Operation is null")`), lookup the handler into a local variable, and if the handler is null throw `new RuntimeException("No handler found for operation: " + operation)`. This avoids downstream NPEs in callers like ShopServiceImpl and makes errors easier to diagnose. Also consider making the exception messages precise and consistent.